nau_openedx_extensions/certificates/context_extender.py

```diff
@@ -84,13 +86,144 @@ def update_context_with_grades(user, course, context, settings, user_certificate
+                        "en": "Regular"
+                    },
+                    "min_included": 50,
+                    "max_excluded": 65
+                },
+                {
+                    "grade_text": {
+                        "pt-pt": "Bom",
+                        "en": "Good"
+                    },
+                    "min_included": 65,
+                    "max_excluded": 80
+                },
+                {
+                    "grade_text": {
+                        "pt-pt": "Muito Bom",
+                        "en": "Very Good"
+                    },
+                    "min_included": 80,
+                    "max_excluded": 90
+                },
+                {
+                    "grade_text": {
+                        "pt-pt": "Excelente",
+                        "en": "Excelent"
+                    },
+                    "min_included": 90,
+                    "max_excluded": 101
+                }
+            ],
+            "grade_round_format": "course_percent_grade:.0%"
+        }
+    },
+
+    """
+    grade_rounded = None
+    try:
+        grade_round_format = course_qualitative_grade_config.get("grade_round_format")
+
+        # prefix with `{` if not already has that character
+        if grade_round_format[0] != "{":
+            grade_round_format = "{" + grade_round_format
+
+        # suffix with `}` if not already has that character
+        if grade_round_format[len(grade_round_format)-1] != "}":
+            grade_round_format += "}"
+
+        grade_rounded = grade_round_format.format(**context)
+        # clear `%` caracter
+        grade_rounded = grade_rounded.replace('%', '')
+    except Exception:  # pylint: disable=broad-except
+        log.error(
+            "Could not round the course grade for qualitative grade scale for user %s in course %s",
+            user.username,
+            course.display_name,
+        )
+
+    # fall back
+    if not grade_rounded:
+        grade_rounded = context.course_percent_grade
+
+    # append to context the rounded grade
+    context.update({"course_grade_rounded" : grade_rounded})
+
+    qualitative_grade = get_qualitative_grade(user, course, certificate_language, course_qualitative_grade_config.get("ranges", []), grade_rounded)
+    if qualitative_grade:
+
+        context.update({"course_grade_qualitative" : qualitative_grade})
+
+
+def get_qualitative_grade(user, course, certificate_language, course_qualitative_ranges_settings, grade_rounded):
+    """
+    Returns a qualitative grade for the rounded grade given the course qualitative ranges settings.
+    """
+    try:
+        grade_rounded_f = float(grade_rounded)
+        for range in course_qualitative_ranges_settings:
+            min_included = float(range.get("min_included", -1.0))
+            max_excluded = float(range.get("max_excluded", -1.0))
+            if min_included <= grade_rounded_f and grade_rounded_f < max_excluded:
```

Comment:
Here is a suggestion to simplify the code
```suggestion
            if min_included <= grade_rounded_f < max_excluded:
```